pandas/_libs/tslibs/conversion.pyx

```diff
@@ -355,6 +366,26 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
         obj.value += nanos
         obj.dts.ps = nanos * 1000
 
+    if tz is not None:
+        if is_utc(tz):
+            pass
+        elif is_tzlocal(tz):
+            pass
+            # TODO: think on how we can infer fold for local Timezone
+            # and adjust value for fold
+        else:
+            trans, deltas, typ = get_dst_info(tz)
+
+            if typ == 'pytz' or typ == 'dateutil':
+                pos = trans.searchsorted(obj.value, side='right') - 1
+                # pytz assumes fold == 1, dateutil fold == 0
+                # adjust only if necessary
+                if ((typ == 'pytz' and fold == 0) or
```

Comment:
Can you remove the outer parenthesis?

Reply:
@mroeschke 
The outer paranthesis are there to allow the multiline if with nice indentation and no ` \` necessary. The inner ones are unnecessary though because of boolean operator priority, so I removed those.
Is that fine?

Reply:
Sure. Perfect